docs/resources/client/textures/atlases.md

```diff
@@ -0,0 +1,98 @@
+
+## Texture Atlas Sources
+
+Texture atlas sources are json files located inside the `atlases` asset folder.
+A texture atlas source describes the name of the targeted atlas and its contents. Additions to the atlases need to be created under the namespace of a mod or datapack.
+
+As of today 14 different atlas sources can be declared:
+
+|File name|Description|
+|--|--| 
+|`blocks`|Block textures|
+|`banner_patterns`|Banner patterns|
+|`beds`|Every bed variant (i.e. colors)|
+|`chest`|Chest textures|
+|`shield_patterns`|Shield textures and patterns|
+|`shulker_boxes`|Every shulker related textures (Shulker box variants, head and projectile)|
+|`signs`|Sign textures|
+|`particles`|Particle textures|
+|`paintings`|Every painting|
+|`mob_effects`|Potion effect icons|
+|`gui`|Every GUI wigets (buttons, scroll bars...)|
+|`map_decorations`|Icons displayed on a map (banners, locations...)|
+|`armor_trims`|Armor trim patterns for every material|
+|`decorated_pot`|Texture for the decorated pot and every sherd|
+
+### Creating an atlas source
+
+To add new textures to an atlas, a new file needs to be created in the `atlases` asset folder: `resources/assets/example_mod/atlases`.
+
+The file named the same as the altas it is appending to, will then describe what file will be added to the atlas:
+```json
+-- Inside gui.json
+{
+  "sources": [
+    -- Gather all sprites within the "source" folder
+    {
+      "type": "minecraft:directory",
+      "prefix": "",
+      "source": "gui/sprites/widgets"
+    },
+    -- Gather sprites within the "icons" spritesheet
+    {
+      "type": "minecraft:unstitch",
+      "resource": "example_mod:gui/sprites/icons",
+      "divisor_x": 0.0,
+      "divisor_y": 0.0,
+      "regions": [
+        {
+          "sprite": "example_mod:energy_low",
+          "x": 0.0,
+          "y": 0.0,
+          "width": 8.0,
+          "height": 8.0
+        }
+      ]
+    }
+  ]
+}
+```
+
+All paths are realtive to the texture folder and extentions (`.png`, `.png.mcmeta`) are implied.
+
+
+## Datagen
+
+Like many resources in Minecraft, texture atlas sources can be proceduraly generated using [datagen][dtgn]. To do so, we extend `SpriteSourceProvider` and override the `gather()` method:
+
+```Java
+public class MySpriteSourceProvider extends SpriteSourceProvider {
+
+  // Parameters obtained via GatherDataEvent
+  public MySpriteSourceProvider (
+    PackOutput output,
+    CompletableFuture<HolderLookup.Provider> lookupProvider
+  ) {
+    super(output, lookupProvider, "example_mod");
+  }
+
+  @Override
+  protected void gather ()
+  {
+    // Creates a new "gui" atlas for the mod's namespace
+    SourceList gui = atlas(
+      ResourceLocation.fromNamespaceAndPath("example_mod", "gui")
+    );
+
+    // Adds the "resources/assets/example_mod/textures/gui/sprites" folder
+    // as a source for the atlas
+    sources.addSource(new DirectoryLister("gui/sprites", ""));
+  }
+}
+```
```

Comment:
See other examples on how we show data generation. Usually, we make it a tab-like system and show how to register it via gather data.